### DISCUSSION TITLE
v2.2.0

### DISCUSSION THREAD

**Discussion by tomtau** [pest-parser/pest#669] [Announcements]
## What's Changed
- Unicode 14 support
- sha-1 upgraded and unnecessary dependencies removed
- code quality improvements and migration to the 2018 edition
- benchmark code migrated to criterion
- optimizations (see https://github.com/pest-parser/pest/pull/473 and https://github.com/pest-parser/pest/pull/554 )
- no_std compatibility
- error reporting improvements (using thiserror, better printing, API improvements)
- allowing infix operators in prefix expressions

## New Contributors
* @yamafaktory made their first contribution in https://github.com/pest-parser/pest/pull/361
* @cch123 made their first contribution in https://github.com/pest-parser/pest/pull/368
* @adamAndMath made their first contribution in https://github.com/pest-parser/pest/pull/371
* @Nadrieril made their first contribution in https://github.com/pest-parser/pest/pull/375
* @golddranks made their first contribution in https://github.com/pest-parser/pest/pull/395
* @PvdBerg1998 made their first contribution in https://github.com/pest-parser/pest/pull/403
* @alex made their first contribution in https://github.com/pest-parser/pest/pull/406
* @roblabla made their first contribution in https://github.com/pest-parser/pest/pull/410
* @cr0sh made their first contribution in https://github.com/pest-parser/pest/pull/408
* @Schaeff made their first contribution in https://github.com/pest-parser/pest/pull/404
* @shnewto made their first contribution in https://github.com/pest-parser/pest/pull/422
* @bookmoons made their first contribution in https://github.com/pest-parser/pest/pull/412
* @gkelly made their first contribution in https://github.com/pest-parser/pest/pull/435
* @nagisa made their first contribution in https://github.com/pest-parser/pest/pull/442
* @pr2502 made their first contribution in https://github.com/pest-parser/pest/pull/445
* @jakubadamw made their first contribution in https://github.com/pest-parser/pest/pull/465
* @MarinPostma made their first contribution in https://github.com/pest-parser/pest/pull/454
* @SkiFire13 made their first contribution in https://github.com/pest-parser/pest/pull/480
* @teymour-aldridge made their first contribution in https://github.com/pest-parser/pest/pull/470
* @MaxBondABE made their first contribution in https://github.com/pest-parser/pest/pull/486
* @rafael-patronilo made their first contribution in https://github.com/pest-parser/pest/pull/468
* @01mf02 made their first contribution in https://github.com/pest-parser/pest/pull/498
* @lucperkins made their first contribution in https://github.com/pest-parser/pest/pull/487
* @mitnk made their first contribution in https://github.com/pest-parser/pest/pull/511
* @nkreipke made their first contribution in https://github.com/pest-parser/pest/pull/514
* @kaikalii made their first contribution in https://github.com/pest-parser/pest/pull/534
* @mrtnzlml made their first contribution in https://github.com/pest-parser/pest/pull/532
* @huacnlee made their first contribution in https://github.com/pest-parser/pest/pull/529
* @tmandry made their first contribution in https://github.com/pest-parser/pest/pull/522
* @Turbo87 made their first contribution in https://github.com/pest-parser/pest/pull/537
* @aofdev made their first contribution in https://github.com/pest-parser/pest/pull/538
* @NoahTheDuke made their first contribution in https://github.com/pest-parser/pest/pull/556
* @Anton-4 made their first contribution in https://github.com/pest-parser/pest/pull/561
* @andybest made their first contribution in https://github.com/pest-parser/pest/pull/562
* @Geobert made their first contribution in https://github.com/pest-parser/pest/pull/575
* @bryantbiggs made their first contribution in https://github.com/pest-parser/pest/pull/585
* @jwpjrdev made their first contribution in https://github.com/pest-parser/pest/pull/592
* @tomtau made their first contribution in https://github.com/pest-parser/pest/pull/609
* @bobbbay made their first contribution in https://github.com/pest-parser/pest/pull/641
* @abhimanyu003 made their first contribution in https://github.com/pest-parser/pest/pull/567

**Full Changelog**: https://github.com/pest-parser/pest/compare/v2.1.0...v2.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/pest-parser/pest/releases/tag/v2.2.0'>v2.2.0</a>.</em>

**Comment by Geobert** [comment]
Thank you very much for the hardwork and keeping `pest` alive!

**Comment by MarijnS95** [comment]
Could https://github.com/pest-parser/pest/pull/522 have been a breaking change? I'm now seeing this with `liquid-core`:

```rust
error: couldn't read /home/marijn/.cargo/registry/src/github.com-1ecc6299db9ec823/liquid-core-0.24.1/src/parser/parser/grammar.pest: No such file or directory (os error 2)
  --> /home/marijn/.cargo/registry/src/github.com-1ecc6299db9ec823/liquid-core-0.24.1/src/parser/parser.rs:19:14
   |
19 |     #[derive(Parser)]
   |              ^^^^^^
   |
   = note: this error originates in the macro `include_str` (in Nightly builds, run with -Z macro-backtrace for more info)
```

When changing `#[grammar = "parser/grammar.pest"]` into `#[grammar = "grammar.pest"]`, a different error appears which I believe comes from:

https://github.com/pest-parser/pest/blob/99cbc31419551af9cd377476382e85c43837daee/generator/src/lib.rs#L52

```rust
error: proc-macro derive panicked
  --> /home/marijn/.cargo/registry/src/github.com-1ecc6299db9ec823/liquid-core-0.24.1/src/parser/parser.rs:19:14
   |
19 |     #[derive(Parser)]
   |              ^^^^^^
   |
   = help: message: error opening "grammar.pest": No such file or directory (os error 2)
```

Is this something that should be updated on the `liquid-core` side, or is the generator taking in the wrong file?

**Reply by NoahTheDuke** [reply]
See #672 as well. Might need to undo this change.

**Reply by MarijnS95** [reply]
Please revert if you can; I've been looking into the linked PR #522 and it doesn't seem correct at all. `include_str!` receives a path relative to the source file yet the same path is used directly on top of `$CARGO_MANIFEST_DIR/src/<the file>`, disregarding any nested modules.

**Comment by tomtau** [comment]
FYI v2.2.0 was yanked and v2.2.1 was published (it reverted the breaking change): https://github.com/pest-parser/pest/releases/tag/v2.2.1

**Reply by tomtau** [reply]
There was another small breaking change found: https://github.com/pest-parser/pest/issues/693 , hence v2.2.1 was yanked and v2.3.0  was published: https://github.com/pest-parser/pest/discussions/701
